iroh-net/src/relay/http/server.rs

```diff
@@ -320,13 +378,17 @@ impl ServerState {
                     }
                 }
             }
+            if let Some(server) = server {
+                // TODO: if the task this is running in is aborted this server is not shut
```

Comment:
what about this todo?

Reply:
I decided to leave that in because I felt like if I had to turn all of the relay server actors into structured concurrency the rabbit hole was too deep.  This is basically the status quo so not a regression.  It would be nice to do this, but as a separate PR probably.

Plus, would need to decide on how much I like tokio-graceful-shutdown possibly :)